### DISCUSSION TITLE
New Ethernet MAC on every reboot (?)

### DISCUSSION THREAD

**Discussion by tak-21** [Guilouz/Klipper-Flsun-Speeder-Pad#121] [Q&A] [answered]
Hi @ all,

I soldered an ethernet port to the board and I now have a working LAN connection which works well.
But ... the MAC address changes on every reboot. I installed macchanger and it is telling me:

`root@v400-red:/home/pi# macchanger -s eth0`
`Current MAC:   4e:fe:f5:28:e4:23 (unknown)`
`Permanent MAC: 00:00:00:00:00:00 (XEROX CORPORATION)`

Why is the Permanent MAC 00:...:00 ? 
I found this thread for armbian: https://forum.armbian.com/topic/14525-mac-address-of-eth0-changes-on-every-boot/ 
Is it possibly the same topic? 

In short: do you know a way to fix it? I would really like to have a static MAC address to then assign a static IP via my DHCP server.

Thanks in advance, Thomas

**Comment by tak-21** [comment, marked as answer]
Just to answer my own question, this helped me.

I updated / replaced /etc/netplan/01-network-manager-all.yaml:

`network:`
`        version: 2`
`        ethernets:`
`                eth0:`
`                        macaddress: ba:e4:64:bc:81:d7`
`                        dhcp4: yes`

And I learned that the first octet is important when you want to create a valid, local mac address. In the example above, the "a". (https://en.wikipedia.org/wiki/MAC_address)
So with this my V400 behaves nicely with DHCP.
